tests/subsys/bluetooth/services/ble_bas/src/unity_test.c

```diff
@@ -0,0 +1,450 @@
+	ble_bas_on_ble_evt(&evt, &ble_bas);
+	TEST_ASSERT_TRUE(evt_handler_called);
+
+	evt_handler_called = false;
+	evt.evt.gatts_evt.params.write.data[0] = 0x00;
+	bas_cfg.evt_handler = ble_bas_evt_handler_notif_disable;
+	bas_init(&bas_cfg);
+	ble_bas_on_ble_evt(&evt, &ble_bas);
+	TEST_ASSERT_TRUE(evt_handler_called);
+
+	evt_handler_called = false;
+	evt.evt.gatts_evt.params.write.data[0] = BLE_GATT_HVX_NOTIFICATION;
+	bas_cfg.can_notify = false;
+	bas_cfg.evt_handler = ble_bas_evt_handler;
+	bas_init(&bas_cfg);
+	ble_bas_on_ble_evt(&evt, &ble_bas);
+	TEST_ASSERT_FALSE(evt_handler_called);
+
+	bas_cfg.can_notify = true;
+	bas_cfg.evt_handler = ble_bas_evt_handler;
+	evt.evt.gatts_evt.params.write.handle = INVALID_HANDLE;
+	bas_init(&bas_cfg);
+	ble_bas_on_ble_evt(&evt, &ble_bas);
+	TEST_ASSERT_FALSE(evt_handler_called);
+
+	evt.evt.gatts_evt.params.write.handle = CCCD_HANDLE;
+	evt.evt.gatts_evt.params.write.len = 1;
+	bas_init(&bas_cfg);
+	ble_bas_on_ble_evt(&evt, &ble_bas);
+	TEST_ASSERT_FALSE(evt_handler_called);
+}
+
+void test_ble_bas_init_efault(void)
+{
+	int ret;
+	struct ble_bas_config bas_config = { .evt_handler = ble_bas_evt_handler };
+
+	ret = ble_bas_init(NULL, &bas_config);
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+	TEST_ASSERT_NOT_EQUAL(bas_config.evt_handler, ble_bas.evt_handler);
+
+	ret = ble_bas_init(&ble_bas, NULL);
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+	TEST_ASSERT_NOT_EQUAL(bas_config.evt_handler, ble_bas.evt_handler);
+}
+
+void test_ble_bas_init_einval(void)
+{
+	int ret;
+	struct ble_bas_config bas_config = { .evt_handler = ble_bas_evt_handler };
+
+	bas_config.report_ref = (void *)&report_ref;
+
+	__cmock_sd_ble_gatts_service_add_ExpectAnyArgsAndReturn(NRF_ERROR_INVALID_PARAM);
+	ret = ble_bas_init(&ble_bas, &bas_config);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+
+	__cmock_sd_ble_gatts_service_add_ExpectAnyArgsAndReturn(NRF_SUCCESS);
+	__cmock_sd_ble_gatts_characteristic_add_ExpectAnyArgsAndReturn(NRF_ERROR_INVALID_PARAM);
+	ret = ble_bas_init(&ble_bas, &bas_config);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+
+	__cmock_sd_ble_gatts_service_add_ExpectAnyArgsAndReturn(NRF_SUCCESS);
+	__cmock_sd_ble_gatts_characteristic_add_ExpectAnyArgsAndReturn(NRF_SUCCESS);
+	__cmock_sd_ble_gatts_descriptor_add_ExpectAnyArgsAndReturn(NRF_ERROR_INVALID_PARAM);
+	ret = ble_bas_init(&ble_bas, &bas_config);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+}
+
+void test_ble_bas_init_success(void)
+{
+	int ret;
+	struct ble_bas_config bas_cfg = bas_cfg_template;
+
+	bas_cfg.evt_handler = ble_bas_evt_handler;
+
+	__cmock_sd_ble_gatts_service_add_Stub(stub_sd_ble_gatts_service_add_success);
+	__cmock_sd_ble_gatts_characteristic_add_Stub(stub_sd_ble_gatts_characteristic_add_success);
+	__cmock_sd_ble_gatts_descriptor_add_Stub(stub_sd_ble_gatts_descriptor_add_success);
+
+	ret = ble_bas_init(&ble_bas, &bas_cfg);
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(SERVICE_HANDLE, ble_bas.service_handle);
+	TEST_ASSERT_EQUAL(VALUE_HANDLE, ble_bas.battery_level_handles.value_handle);
```

Comment:
Here too, you can call the `ble_bas_battery_level_notify()` function to see that the `hvx` parameters have the correct handle for the value.